crates/bevy_render/src/lib.rs

```diff
@@ -189,9 +189,10 @@ pub mod main_graph {
     }
 }
 
+/// A temporary resource representing renderer resources that may still be in the process of being created.
 #[derive(Resource)]
-struct FutureRendererResources(
-    Arc<
+pub struct FutureRendererResources(
```

Comment:
What would modifying this resource at runtime do? 

If nothing or it would break things, would it make more sense to add these as an optional field to the RenderApp that RenderApp would use to init this resource instead? Seems like that might be a better way to go for one time configuration?

Reply:
it would be very unlikely to break anything. the only time that resource ever matters is during the plugin build stage. If a plugin replaces that resource after the render crate has created it, nothing bad will happen except it will use the plugin's render resources. i just recommend initializing them before because it will let the render crate skip wgpu initialization.

Reply:
Cool, so if they only do anything at initialization I'd probably recommend not exposing the resource (which could have been modified at any time) and instead allow them to be provided as optional fields when initializing the plugin. 

See the asset plugin as an example of this https://docs.rs/bevy/latest/bevy/asset/struct.AssetPlugin.html you can provide optional watch for changes settings.

For why this is better, 
- it makes it clearer to users how to use it just by looking at the API because it isn't possible to set it at the wrong time.
- prevents multiple third party crates from trying to all create and set it without the user realizing.
- makes it more obvious to the user that this is something that can be configured with the render app.

Reply:
mhm, how do you think i should implement it? should i add a `Option<FutureRenderResources>` field to the render plugin, or should i make that struct private again and have people pass the render resources to the render plugin? i think i should go with the first option, since the render resources aren't actually initialized until the plugin build phase.

Reply:
Hmm not sure, I was leaning towards it just being a field that takes an option with a tuple, or multiple fields if it makes sense to only provide some of them sometimes. I haven't really dug into the way they get initialized or anything so idk if multiple fields would make sense. I would lean towards not exposing a resource so people don't think they can do something with that resource. Usually when I see a public resource I assume I can do something with it at runtime. I'll leave that up to you though, I try not to nit pick too much.

Also, this was just a thought I had while reviewing and I'm not really anyone particularly important here lol. Feel free to disagree after considering it.
Looking forward to more VR Bevy stuff so I don't mean to hold things up!

Reply:
An optional field on RenderPlugin would be nice. Definitely no public resource if it can't be set at runtime.

Similarly, it might be helpful to allow specifying only some of these settings if partial initalization is meaningful.

Reply:
right now the problem i'm having though is that usually the render resources aren't actually initialized until the build phase of the plugin, however, having them as an optional field on the RenderPlugin would require the render resources to be already initialized when the plugin is made

Reply:
I was thinking that for a third party crate to set these you would have users disable Bevy's render plugin from DefaultPlugins and then add the third party's render plugin which would simply init the render resources and then add Bevy's render plugin back with those resources in the third party plugin's build step.

The third party crate could also provide their own DefaultPlugins that does something similar if they want to simplify what the user has to do a bit.

Would that work?

Reply:
that sounds like it would work perfectly,